src/main/java/io/cdap/plugin/hubspot/common/BaseHubspotConfig.java

```diff
@@ -0,0 +1,266 @@
+    "\n" +
+    "The analytics report type of object that you want the analytics data for. Must be one of:\n" +
+    "\n" +
+    "event-completions - Get data for analytics events. The results are broken down by the event ID." +
+    " You can get the details for the events using this endpoint.\n" +
+    "forms - Get data for your HubSpot forms. The results are broken down by form guids." +
+    " You can get the details for the forms through the Forms API. \n" +
+    "pages - Get data for all URLs with data collected by HubSpot tracking code." +
+    " The results are broken down by URL. \n" +
+    "social-assists - Get data for messages published through the social publishing tools." +
+    " The results are broken down by the broadcastGuid of the messages." +
+    " You can get the details of those messages through the Social Media API.")
+  @Macro
+  @Nullable
+  public String reportType;
+  @Name(TIME_PERIOD)
+  @Description("The time period used to group the data. Must be one of:\n" +
+    "\n" +
+    "total - Data is rolled up to totals covering the specified time.\n" +
+    "daily - Grouped by day\n" +
+    "weekly - Grouped by week\n" +
+    "monthly - Grouped by month\n" +
+    "summarize/daily - Grouped by day, data is rolled up across all breakdowns \n" +
+    "summarize/weekly - Grouped by week, data is rolled up across all breakdowns \n" +
+    "summarize/monthly - Grouped by month, data is rolled up across all breakdowns")
+  @Macro
+  @Nullable
+  public String timePeriod;
+  @Name(APP_ID)
+  @Description("OAuth2 App ID")
+  @Macro
+  public String appId;
+  @Name(API_KEY)
+  @Description("OAuth2 API Key")
+  @Macro
+  public String apiKey;
+  @Name(OBJECT_TYPE)
+  @Description("Select from Contact Lists, Contacts, Email Events, Email Subscription," +
+    " Recent Campaigns, Analytics, Companies, Deals, Deal Pipelines, Marketing Email, Products, Tickets")
+  @Macro
+  public String objectType;
+  @Name(CALLS_PER_DAY)
+  @Description("The number of API calls to make per day, to avoid hitting HubSpot's rate limits")
+  @Macro
+  public int callsPerDay;
+
+  public BaseHubspotConfig(String referenceName) {
+    super(referenceName);
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    validateAuthorization(failureCollector);
+    validateObjectType(failureCollector);
+    if (!containsMacro(OBJECT_TYPE) && getObjectType().equals(ObjectType.ANALYTICS)) {
+      validateReportType(failureCollector);
+      validateTimePeriod(failureCollector);
+      validateFilters(failureCollector);
+      validateDateRange(failureCollector);
+    }
+  }
+
+  private void validateTimePeriod(FailureCollector failureCollector) {
+    if (containsMacro(TIME_PERIOD)) {
+      return;
+    }
+    try {
+      getTimePeriod();
+    } catch (IllegalArgumentException e) {
+      failureCollector.addFailure(String.format("Time Period '%s' is not a valid Time Period", timePeriod),
```

Comment:
`Time period '%s' is not a valid.`,
also add corrective action with the correct format.